src/scikit_build_core/settings/metadata.py

```diff
@@ -53,33 +24,23 @@ def get_standard_metadata(
     metadata = StandardMetadata.from_pyproject(pyproject_dict)
 
     # handle any dynamic metadata
-    # start by collecting all the scikit-build entrypoints
-    eps: importlib.metadata.EntryPoints = importlib.metadata.entry_points(
-        group="scikit_build.metadata"
-    )
-    cached_plugins = {}
-    for field, ep_name in settings.metadata.items():
+    for field in settings.metadata:
         if field not in metadata.dynamic:
             msg = f"{field} is not in project.dynamic"
             raise KeyError(msg)
```

Comment:
I am not sure it is better to split this up like this, it just means we loop over the `settings.metadata` dict twice (and the plugins set as well), and is not actually any more compact if we drop the try-except on missing plugins. Not a major time issue with the number of plugins we will be concerned with, so I am happy to accept this alternative formulation, but for my own edification I just wondered if there was a good reason that I am missing here.

Reply:
This allows you to process the plugins once, so that if a plugin is listed twice, it is only called once. You could build the set inside the first loop, I suppose, but dynamically adding one at a time to a set is a bit less elegant than a set comprehension.